cmake/AwsCFlags.cmake

```diff
@@ -8,6 +8,7 @@ include(CMakeParseArguments) # needed for CMake v3.4 and lower
 option(AWS_ENABLE_LTO "Enables LTO on libraries. Ensure this is set on all consumed targets, or linking will fail" OFF)
 option(LEGACY_COMPILER_SUPPORT "This enables builds with compiler versions such as gcc 4.1.2. This is not a 'supported' feature; it's just a best effort." OFF)
 option(AWS_SUPPORT_WIN7 "Restricts WINAPI calls to Win7 and older (This will have implications in downstream libraries that use TLS especially)" OFF)
```

Comment:
I vote we turn this OFF by default, and fast-follow this PR with changes to our CI so it sets the flag.


I just think philosophically, a default value should be aimed at an external user. If an option is there to help developers of THIS library, then it should be opt in.

Reply:
I want to avoid the window of disabling warnings as errors for downstream consumers, and not have to preemptively add turning it on in those consumers.

Reply:
We can tackle this in the future (I just want that future now 😢)